[Bugfix] Updating default timeout

akashsinghal opened the pull request:
Fixes #92 

Root Cause:

- The Gatekeeper validating webhook timeout is set to 3 seconds while the current verification timeout is 8 seconds
- A longer running verification process will cause the Gatekeeper webhook to timeout and cancel the context
- The error will be emitted but code structure will short circuit to success. 

Solution: 
- Set timeout to be less than or equal to the webhook timeout (3 seconds). Verifier will timeout before webhook does and block the deployment as expected. 

Validation Steps:
1. Update notaryv2 plugin to sleep for time > 3 seconds
2. Attempt to deploy signed image. 
3. Deployment is blocked with error message: `Error from server ([ratify-constraint] Image verification failed : {"errors": [], "responses": [], "status_code": 200, "system_error": "validate operation failed with error operation timed out after duration 3s"}): admission webhook "validation.gatekeeper.sh" denied the request: [ratify-constraint] Image verification failed : {"errors": [], "responses": [], "status_code": 200, "system_error": "validate operation failed with error operation timed out after duration 3s"}`

sajayantony reviewed (changes requested):
This needs a test so that this doesn't regress.

akashsinghal commented:
I added a few more changes:
1. Converted the timeout to use 2800 milliseconds so there's some buffer of the 3 second webhook timeout limit.
2. Added a `respCode` parameter to the `SendResponse` method to easily send different status codes based on errors/success
3. Edited `TestVerifiers` struct so its parameters are exportable and can be used by the server test
4. Added a server test that will check request times out if verification takes too long. Currently returns a 500 error if server times out. Not sure if we need to change the error codes. /cc: @sajayantony

etrexel reviewed (approved):
LGTM
